exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/VarLenColumnBulkInput.java

```diff
@@ -479,6 +483,17 @@ private void deinitOverflowData() {
     fieldOverflowStateContainer = null;
   }
 
+  private void checkCancellation() {
+    // - The methods hasNext() & next() are usually invoked within a loop
+    // - We need to ensure that bugs do not cause such loops to execute forever
+    // - To do that, will check the interrupted status (e.g., cancellation) to break the infinite loop
+    if (Thread.currentThread().isInterrupted()) {
```

Comment:
use `Thread.interruped()`

Reply:
@vrozov,
- Thread.interruped() will clear the thread interrupted status; I want to propagate the thread interruption status till the higher level framework code clears it to state "ok; received the cancellation request and now ready to handle other requests"
- Are you suggesting that it is not necessary to still continue setting this flag since we are throwing an exception?

please clarify!

Reply:
It is not necessary to keep the flag after the exception is thrown. It will be better to throw InterruptedException outside of Iterator.